app/graphql/types/query_type.rb

```diff
@@ -118,6 +118,18 @@
     end)
   end
 
+  field :reasoningAttempt, Types::ReasoningAttemptType do
+    description 'A ReasoningAttempt'
+
+    argument :id, !types.Int do
+      description 'The id of the ReasoningAttempt'
+    end
+
+    resolve(lambda do |_root, arguments, _context|
```

Comment:
It cannot be reached because it's only nested inside repository, but you are right. We should call `authorize` again to be sure. Maybe we change the nesting some day and forget to `authorize`.

Reply:
Wait... this is not always nested in repository. It *needs* `authorize`.

Reply:
This needs some work (additional associations) in the models...